### PR TITLE
PWGHF: Add D+ and Ds+ to MC efficiency task

### PR DESCRIPTION
**Pull request by fgrosa**
@jgrosseo @mfaggin I have added the possibility to run the MC efficiency task with D+ and Ds. Could you please have a look if the implementation is fine with you? Thanks a lot!

### CODE REVIEW

File: PWGHF/Tasks/taskMcEfficiency.cxx [hunk 1]
```diff
@@ -177,12 +199,18 @@ struct HfTaskMcEfficiency {
...
+          } else if (pdgCode == pdg::kDS) {
+            massHypo1 = invMassDsToKKPi(candidate);
           }
           hCandidates->Fill(kHFStepTracked, pt, massHypo1, pdgCode, cpa, collisionMatched, origin);
         }
         if (isHypoMass2TrackStep) {
           if (pdgCode == pdg::kLambdaCPlus) {
             massHypo2 = invMassLcToPiKP(candidate);
+          } else if (pdgCode == pdg::kDS) {
+            massHypo1 = invMassDsToPiKK(candidate);
```
Comment (mfaggin) [inline PWGHF/Tasks/taskMcEfficiency.cxx:213]: @fgrosa shouldn't it be `massHypo2` here?

Comment (fgrosa) [reply PWGHF/Tasks/taskMcEfficiency.cxx]: It was a typo indeed, fixed, thanks!

**Review by mfaggin — COMMENTED** [review]
hi @fgrosa from a quick look it seems really good, extending what we thought together. I just have 1 question, see the other comment.

Fine for me for the rest.
Best.
Mattia

**Comment by vkucera** [comment]
@jgrosseo Can you please review this?

**Review by jgrosseo — APPROVED** [review]
This is fine for me. We should see if we find a better way for the additional template arguments (`bool hasDplus, bool hasDs, bool hasLc`) but this is not a showstopper...

**Review by vkucera — APPROVED** [review]

**Pull request merged**